tesseract/integration-test/src/lib.rs

```diff
@@ -392,57 +320,52 @@ async fn get_request_works() -> Result<(), anyhow::Error> {
 	let _ = log_setup();
 	let (chain_a_sub_client, chain_b_sub_client) = create_clients().await?;
 
-	log::info!(" \n 🧊integration test for para: 2000 to para 2001: get request \n");
+	log::info!("🧊integration test for para: 2000 to para 2001: get request \n");
 
 	// =======================================================================
-	let (chain_a_client, chain_b_client) =
-		(Arc::new(chain_a_sub_client.clone()), Arc::new(chain_b_sub_client.clone()));
+	let chain_b_client = Arc::new(chain_b_sub_client.clone());
 
-	let (client_a, client_b) =
-		(chain_a_sub_client.clone().client, chain_b_sub_client.clone().client);
-
-	// Accounts & keys
-	let dave_signer = PairSigner::<Hyperbridge, _>::new(
-		Pair::from_string("//Dave", None).expect("Unable to create Dave account"),
-	);
 	// parachain info pallet fetching para id
 	let encoded_chain_b_id_storage_key =
 		"0x0d715f2646c8f85767b5d2764bb2782604a74d81251e398fd8a0a4d55023bb3f";
+	let key = hex::decode(encoded_chain_b_id_storage_key.strip_prefix("0x").unwrap()).unwrap();
 
-	let latest_height_b =
-		chain_a_client.query_latest_height(chain_b_client.state_machine_id()).await? - 5;
+	let latest_height_b = chain_b_client.query_finalized_height().await?;
 
-	let get_request = api::tx().ismp_demo().get_request(GetRequest {
+	let get_request_param = IsmpPalletDemo::GetRequest {
 		para_id: 2001,
-		height: latest_height_b,
+		height: latest_height_b as u32,
 		timeout: 0,
-		keys: vec![hex::decode(encoded_chain_b_id_storage_key.strip_prefix("0x").unwrap()).unwrap()],
-	});
+		keys: vec![key.clone()],
+	};
 
-	let tx_result = client_a
-		.tx()
-		.sign_and_submit_then_watch_default(&get_request, &dave_signer)
-		.await?
-		.wait_for_finalized_success()
-		.await?
-		.all_events_in_block()
-		.clone();
+	chain_a_sub_client.get_request(get_request_param).await?;
 
-	let tx_block_hash = tx_result.block_hash();
-	let tx_block_height = client_a.blocks().at(tx_block_hash).await?.number() as u64;
-	let events = client_a.events().at(tx_block_hash).await?;
-	let event = events.find_last::<RequestEvent>()?.unwrap();
-	log::info!("Ismp Events: {:?} \n", event);
+	let event = chain_a_sub_client.pallet_ismp_demo_events_stream(1, "Ismp", "Request").await?;
```

Comment:
```suggestion
	let event = chain_a_sub_client.pallet_ismp_demo_events_stream(1, "Ismp", "Request").await?;
```
I prefer you don't use this function, but rather modify the function that submits the tx to return the block hash where the Tx was included, then fetch events from that block.